### PR TITLE
Fix path metadata

### PR DESCRIPTION
**Pull request by elijahbenizzy**
Fixes path metadata for s3/other URLs.

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: hamilton/io/utils.py [hunk 1]
```diff
@@ -21,12 +24,28 @@ def get_file_metadata(path: str) -> Dict[str, Any]:
     - the last modified time
     - the current time
     """
+    if isinstance(path, Path):
+        path = str(path)
+    parsed = parse.urlparse(path)
+    size = 0
```
Comment (skrawcz) [inline hamilton/io/utils.py:30]: this should be None or -1. 0 could be a real value.

File: tests/io/test_utils.py [hunk 2]
```diff
@@ -16,3 +18,19 @@ def test_get_sql_metadata():
...
+    assert metadata["file_metadata"]["path"] == str(file_path)
+    assert metadata["file_metadata"]["size"] > 0
+    assert metadata["file_metadata"]["last_modified"] == file_path.stat().st_mtime
+    assert metadata["file_metadata"]["timestamp"] is not None
+
+
+def test_get_file_metadata_url_schema():
+    url = "s3://bucket/key"
+    metadata = get_file_metadata(url)
+    assert metadata["file_metadata"]["path"] == url
```
Comment (skrawcz) [inline tests/io/test_utils.py:36]: shouldn't this check scheme?

**Review by skrawcz — APPROVED** [review]
can add test that checks scheme & notes

File: tests/io/test_utils.py (hunk 2 above)
Comment (elijahbenizzy) [reply tests/io/test_utils.py:36]: schema yes, not notes cause that's not really part of the API

**Review by skrawcz — APPROVED** [review]

**Pull request merged**